Docs Refresh (#213)
SleipnirGroup/Choreo: docs/usage/editing-paths.md

> Choreo uses trajectory optimization find the optimal route through waypoints. Additionally, Choreo allows fine-grained customization while still maintaining concrete stability. This allows for smoother paths and better odometry.

Reviewer: It might be nice to mention that you're using "optimization" in the mathematical sense, since many FRC programmers won't have ever heard it used this way.

For example,
```suggestion
Choreo finds the [mathematically optimal](https://en.wikipedia.org/wiki/Mathematical_optimization) trajectory that passes through your waypoints, subject to your drivetrain's constraints. It also allows fine-grained customization while still maintaining concrete stability. This allows for smoother paths and better odometry.
```

## Waypoints

Waypoints are an ordered position that you'd like to hit on your path. There are multiple types of waypoints based on what you'd like to constrain in your path.

To add a waypoint, select the type of waypoint from the waypoints navbar on the top to get started, then click on the field where you want it. Your waypoints will show up on the sidebar. A yellow dot represents the currently selected waypoint, green for the starting waypoint, and red for the ending waypoint.

![waypoints navbar](../media/waypoints+navbar.png)

### Pose Waypoints

![pose waypoint](../media/pose_waypoint.png)

Pose Waypoint is the first button on the waypoints navbar and consists of a translation and a rotation. Drag the little yellow circle (outlined in black) to change the heading.

> - **X and Y**: position in meters from the bottom left (origin)

Reviewer: Should "and" also be bold?

> - **θ**: Angular offset in radians -pi to pi, 0 to right

Reviewer: ```suggestion
- **θ**: Angular offset in radians -π to π, 0 to right
```

### Translation Waypoints

![translation waypoint](../media/translation_waypoint.png)

> Translation Waypoint is the second button on the waypoints navbar and consists of a translation but not a rotation. The UI still includes the yellow circle, but isn't only stored to be able to convert later into a Pose Waypoint. However, rotation is still used as part of the initial guess.

Reviewer: Do you mean "it is" rather than "isn't" in the second sentence? As is this doesn't make sense.

- **X and Y**: position in meters from the bottom left (origin)

### Empty Waypoint

![empty waypoint](../media/empty_waypoint.png)

> The Empty Waypoint allows you to apply a constraint somewhere along a segment without putting a position or heading constraint at that waypoint. (similar to constraints).

Reviewer: Elsewhere this document has been referring to waypoint types without any article.
```suggestion
Empty Waypoint allows you to apply a constraint somewhere along a segment without putting a position or heading constraint at that waypoint. (similar to constraints).
```

> The Empty Waypoint allows you to apply a constraint somewhere along a segment without putting a position or heading constraint at that waypoint. (similar to constraints).

Reviewer: ```suggestion
The Empty Waypoint allows you to apply a constraint somewhere along a segment without putting a position or heading constraint at that waypoint, similar to [Constraints](#constraints).
```

## Initial Guess Point

> The initial guess waypoint just affects the straight-line-between-waypoints gray line, which is used as the starting point for optimization. (Until obstacles are added, this doesn’t have much effect.)

Author: I would but choreo refers to it as such, which would confuse the end user.
<img width="167" alt="image" src="https://github.com/SleipnirGroup/Choreo/assets/70717139/18a4d8fa-c292-4eae-abd0-71a31943073d">

> The initial guess waypoint just affects the straight-line-between-waypoints gray line, which is used as the starting point for optimization. (Until obstacles are added, this doesn’t have much effect.)

Author: maybe something to PR in..

## Constraints

Constraints are limitations that the optimizer needs to respect while generating a path. Different constraints can be applied in different scopes or ranges.

### Waypoint Scope

Applies this constraint between at that specific waypoint.

### Segment Scope

Applies this constraint between waypoints (intuitive example: think constraining your trajectory as a straight line between two waypoints)

### Waypoint + Segment Scope

Like a segment scope, but including segments (intuitive example: think of zero angular velocity)

> | Name                        | Description                                                                                                                                                                                                                                          | Waypoint Scope | Segment Scope |

Author: can you please elaborate on this and the above?

| Max Velocity                | Constrains the max velocity to a specific value throughout multiple waypoints/segments This generally refers to the Maximum Velocity of robot chassis in m/s. A default value can be defined in Robot Config ([Getting Started](../robot-config.md)) | Yes            | Yes           |
| Zero Angular Velocity       | Ensures zero angular velocity through the waypoint and/or segment scope.                                                                                                                                                                             | Yes            | Yes           |
| Straight Line               | Tells the solver to draw a straight line as the path between two immediate waypoints.                                                                                                                                                                | No             | Yes           |

## Obstacles

> Obstacles allow the robot to avoid a certain area of the field throughout the path. For example, in the 2023 Charged Up game, you might want to avoid crashing into the charge station. Obstacles makes it to define where you want the robot to start and end without unnecessary intermediary waypoints.

Reviewer: ```suggestion
Obstacles allow the robot to avoid a certain area of the field throughout the path. For example, in the 2023 Charged Up game, you might want to avoid crashing into the charge station. Obstacles makes it easy to define where you want the robot to start and end without unnecessary (and performance-impacting) intermediary waypoints.
```

To start, select the circle obstacle from the navbar (looks like a circle with a slash through it).

> Then, select anywhere on the field to add an obstacle. You can click on the ends of the circle to resize them.

Author: ill include it as ordered lists

![Obstacles](../media/obstacles.png)

## Generating

> You can create paths by chaining waypoints together. Check out [Controls & Shortcuts](./controls-shortcuts.md) for advanced controls. Once you have at least two waypoints, then you can generate a path. You can cancel the path generation by pressing control + clicking the red X once it has started.

Reviewer: ```suggestion
You can create paths by chaining waypoints together. Check out [Controls & Shortcuts](./controls-shortcuts.md) for advanced controls. Once you have at least two waypoints, then you can generate a path. You can cancel the path generation by pressing control and clicking the red "X" once it has started.
```

> If there is an error generating a path, check to make sure your waypoints can physically work.

Reviewer: Would probably be better as an [admonition](https://squidfunk.github.io/mkdocs-material/reference/admonitions/).
